parlai/core/metrics.py

```diff
@@ -57,53 +65,77 @@ class Metric(ABC):
     @abstractmethod
     def value(self) -> float:
         """
-        Return the value of the metric contained by the metric object, usually a scalar.
-
-        (For instance, if the metric object is SumMetric, .value() will return the sum
-        stored by the object.)
+        Return the value of the metric as a float.
         """
         pass
 
+    def __iadd__(self, other):
+        return self.__radd__(other)
+
+    def __radd__(self, other):
+        if other is None:
+            return self
+        return self.__add__(other)
+
     def __str__(self) -> str:
         return f'{self.value():.4g}'
 
     def __repr__(self) -> str:
-        return f'{self.__class__} ({self.value():.4g})'
+        return f'{self.__class__.__name__}({self.value():.4g})'
 
-    def as_number(self, obj: TScalar) -> Union[int, float]:
+    def __float__(self) -> float:
+        return self.value()
+
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, Metric):
+            return self.value() == other.value()
+        else:
+            return self.value() == other
+
+    def __lt__(self, other: Any) -> bool:
```

Comment:
Do you need to overload greater-than as well?

Reply:
no, the [total_ordering](https://docs.python.org/3/library/functools.html#functools.total_ordering) decorator fills out the rest.